Sources/Swift/Integrations/UserFeedback/SentryUserFeedbackIntegrationDriver.swift

```diff
@@ -39,59 +35,131 @@ class SentryUserFeedbackIntegrationDriver: NSObject, SentryUserFeedbackWidgetDel
+        || !config.location.contains(.trailing) && config.location.contains(.leading)
         let noOpposingVerticals = config.location.contains(.top) && !config.location.contains(.bottom)
-            || !config.location.contains(.top) && config.location.contains(.bottom)
+        || !config.location.contains(.top) && config.location.contains(.bottom)
         let atLeastOneLocation = config.location.contains(.trailing)
-            || config.location.contains(.leading)
-            || config.location.contains(.top)
-            || config.location.contains(.bottom)
+        || config.location.contains(.leading)
+        || config.location.contains(.top)
+        || config.location.contains(.bottom)
         let notAll = !config.location.contains(.all)
         let valid = noOpposingVerticals && noOpposingHorizontals && atLeastOneLocation && notAll
-        #if DEBUG
+#if DEBUG
         assert(valid, "Invalid widget location specified: \(config.location). Must specify either one edge or one corner of the screen rect to place the widget.")
-        #endif // DEBUG
+#endif // DEBUG
         if !valid {
             SentryLog.warning("Invalid widget location specified: \(config.location). Must specify either one edge or one corner of the screen rect to place the widget.")
         }
     }
-    
-    // MARK: SentryUserFeedbackWidgetDelegate
-    
-    func capture(feedback: SentryFeedback) {
-        delegate?.capture(feedback: feedback)
+
+    func observeScreenshots() {
+        if configuration.showFormForScreenshots {
+            NotificationCenter.default.addObserver(self, selector: #selector(userCapturedScreenshot), name: UIApplication.userDidTakeScreenshotNotification, object: nil)
+        }
+    }
+
+    @objc func userCapturedScreenshot() {
+        stopObservingScreenshots()
+        let image = screenshotProvider.appScreenshots().first
+        guard let presenter = presenter else {
+            SentryLog.warning("No UIViewController available to present form.")
+            return
+        }
+        showForm(presenter: presenter, screenshot: image)
+    }
+
+    func stopObservingScreenshots() {
+        NotificationCenter.default.removeObserver(self, name: UIApplication.userDidTakeScreenshotNotification, object: nil)
+    }
+
+    var presenter: UIViewController? {
+        if let customButton = configuration.customButton {
+            return customButton.controller
+        }
+        
+        return widget?.rootVC
+    }
+}
+
+extension UIView {
+    var controller: UIViewController? {
+        var responder = next
+        while responder != nil {
+            guard let resolvedResponder = responder else { break }
+            let klass = type(of: resolvedResponder)
+            guard klass.isSubclass(of: UIViewController.self) else {
+                responder = resolvedResponder.next
+                continue
+            }
+            return resolvedResponder as? UIViewController
+        }
+        return nil
     }
```

Comment:
`m`: Please add somme comments why the next responder lookup is implemented like this. Is there no internal implementation to find the next responder? Why is the method called `controller` which I would expect to be the controller managing this view? Some additional context could be helpful down the line to maintain this SDK.

Reply:
> Please add somme comments why the next responder lookup is implemented like this. 

I'll add it as a comment, but to explain here, it's because we need the UIViewController instance that owns the view in question, which will either be a custom button or our own widget button, so that we can call `presentViewController` to display the form. 

> Is there no internal implementation to find the next responder? 

Not that I know of, or I would've used it, but someone can feel free to point one out if there is one.

> Why is the method called controller which I would expect to be the controller managing this view? 

It is returning the deepest nested UIViewController, see my comment above. I will rename it to `viewController`.